src/routes/communityGroups/community_groups.contoller.js

```diff
@@ -0,0 +1,82 @@
+const fs = require("fs");
+const { promisify } = require("util");
+
+const { ERROR_CODES } = require("../../middleware/error-handler");
+
+const data = require("../../../db/communityGtoups.data.json");
+
+const writeFile = promisify(fs.writeFile);
+
+const dbPath = "db/communityGtoups.data.json";
+
+// return all community groups
+const listCommunityGroups = (req, res) => {
+  return res.status(200).json(data);
+};
+
+// get a community group by its id if it exists
+const getCommunityGroup = (req, res) => {
+  const record = data.filter(
+    record => record.id === parseInt(req.params.id)
+  )[0];
+
+  if (!record) {
+    throw new Error(
+      JSON.stringify({
+        status: ERROR_CODES.NOT_FOUND,
+        message: "Record not found"
+      })
+    );
+  }
+
+  return res.status(200).json(record);
+};
+
+// add a community group if the request has no errors
+const addCommunityGroup = async (req, res) => {
+  const id = data.length + 1;
+  const newCommunityGroupsData = [...data, { id, ...req.body }];
+  await writeFile(dbPath, JSON.stringify(newCommunityGroupsData));
+  return res.status(201).json({
+    id,
+    ...req.body
+  });
+};
+
+// update a communityGroup by its id if the request has no errors
+const updateCommunityGroup = async (req, res) => {
+  let groupData;
+
+  const newCommunityGroupsData = data.map(group => {
```

Comment:
There is a much faster way to update a single record in a set of records like this, but it involves some pre-processing.  Do you know what it is?

Reply:
The immediate solution that comes to mind would be transforming the array into an object, with `id` properties, so we could access records directly by their id.
Like
```
{
  "0": {
    "name": "Murphy - Wyman",
    "description": "A group that cares about Cheese",
    "location": "Andersonport, Qatar"
  }
}
```

I'm not sure that'd be more efficient though.
Right now, no other methods come to mind.